demo/tests/HeaderComponentUITests.swift

```diff
@@ -80,18 +81,38 @@ class HeaderComponentUITests: UITestCase {
         let header = XCUIApplication().otherElements["header"]
         XCTAssertTrue(header.exists)
         
-        // Go to the top of the view
-        (0..<numberOfSwipes + 1).forEach { _ in
+        // Go to the top of the view and make sure that the header is now uncollapsed
+        execute(block: {
             collectionView.swipeDown()
-        }
-        
-        // Make sure that the header is now uncollapsed
-        XCTAssertEqual(header.frame.height, 250)
+        }, until: header, hasHeight: 250)
     }
     
     // MARK: - Utilities
     
     private func navigateToStickyHeaderFeature() {
         XCUIApplication().collectionViews.staticTexts["Sticky header"].tap()
     }
+
+    private func execute(block:() -> Void, until header:XCUIElement, hasHeight expectedHeight:CGFloat) {
```

Comment:
It's usually preferred if you can put the block as the last parameter. This will have two benefits:
1. The call site will be a lot easier to read with:
``` swift
execute(until: header, hasHeight: 250, block: ^{})
```

2. When the block is the last parameter you can even leave out that as an argument when calling:
``` swift
execute(until: header, hasHeight: 250) {
    collectionView.swipeDown()
}
```